src/snowcli/cli/nativeapp/manager.py

```diff
@@ -319,23 +331,36 @@ def app_exists(self) -> bool:
         """Returns True iff the application exists on Snowflake."""
         with self.use_role(self.app_role):
             show_app_cursor = self._execute_query(
-                f"show applications like '{self.app_name}'", cursor_class=DictCursor
+                f"show applications like '{unquote_identifier(self.app_name)}'",
+                cursor_class=DictCursor,
+            )
+            return (
+                find_row(
+                    show_app_cursor,
+                    lambda row: row[NAME_COL] == unquote_identifier(self.app_name),
```

Comment:
Should we make this into a utility function instead of defining it twice in lines 291 and 340?

Reply:
I debated it. One thing I didn't want to do was hide the fact that we were unquoting the identifier. We could do something like:
```python
find_row_matching(show_app_cursor, name=unquote_identifier(self.app_name))
```
But maybe that's too magical (unpacking kwargs as things that need to match in the dict)? What do you think?

Reply:
I tried it out; it doesn't really help with readability and it makes it harder to use a constant for `NAME_COL`, so I think we should keep things as-is.